server/mongodb/actions/UserManager.ts

```diff
@@ -14,6 +15,17 @@ export async function getUserByEmail(email: string) {
   return UserDocument.findOne({ email: email });
 }
 
+export async function upsertUserByProviderProfile(
```

Comment:
This can probably merged with some of the other user functions. For now, to minimize changes in this PR, I'll keep it separate